operator/pkg/lbipam/lbipam.go

```diff
@@ -576,6 +576,26 @@ func (ipam *LBIPAM) stripInvalidAllocations(sv *ServiceView) error {
 	return errs
 }
 
+func (ipam *LBIPAM) checkSharingGroupCompatibility(sv *ServiceView) bool {
```

Comment:
@viktor-kurchenko it seems as only the new function has been added - without changing the code that should actually use the new function.

Reply:
Sorry my bad.
I've pushed the fix and added missing [annotation](https://github.com/cilium/cilium/pull/34456/files#diff-b973f44f6ee5c61824c7ae5d965ff9d9bbf8f94f3a69a83f703b2dbd9762517bR34-R147) required for the [test](https://github.com/cilium/cilium/pull/34456/files#diff-3ae60285c2fcc391862c04ee16909c917ccc3dbedd2efb9a00b1f641a1a7ae1aR1278).
Please take a look, thanks!

Reply:
Backporting the new k8s annotation aliases for the tests only isn't an option, as it seems that these aliases aren't supported by the LB IPAM logic on `v1.15`. Therefore we should use the "old style" annotation keys instead - otherwise the unit-tests are failing (even locally with `go test ./operator/pkg/lbipam/...`_

Applying this patch should do it (at least local tests are working) [patch.txt](https://github.com/user-attachments/files/16660902/patch.txt)

Reply:
Done, thank you!

Reply:
Thanks, looking good :tada:  sorry for the inconvenience!